Documentation/DesignRationale.md

```diff
@@ -0,0 +1,195 @@
+    return rhs(lhs)
+}
+```
+
+This enables us to write
+
+```swift
+a >- map { $0 * 2 } >- filter { $0 > 0 }
+```
+
+instead of
+
+```swift
+a.map { $0 * 2 }.filter { $0 > 0 }
+```
+
+This is another explanation:
+
+```swift
+a >- b >- c is equivalent to c(b(a))
+```
+
+So why was this introduced and not just use "." and extensions? Short answer is that Swift extensions weren't powerful enough, but there are other reasons as well.
+
+Next version of RxSwift for Swift 2.0 will probably also include extensions that will enable the use of
+`.`.
+
+">-" also enables us to chain results easily. For example, if using protocol extensions typical example would look like this.
+
+```swift
+disposeBag.addDisposable(
+    observable
+      .map { n in
+          n * 2
+      }
+      .subscribeNext { n in
+          print(n)
+      }
+  )
+```
+
+This code could be written more elegantly using `>-` operator.
+
+```swift
+observable
+    >- map { n in
+        n * 2
+    }
+    >- subscribeNext { n in
+        print(n)
+    }
+    >- disposeBag.addDisposable
+```
+
+All of the Rx public interfaces don't depend at all on the `>-` operator.
+
+It was actually introduced quite late and you can use Rx operators (map, filter ...) without it.
+
+If you dislike `>-` operator and want to use `|>` or `~>` operators, just define them in your project in this form:
+
+```swift
+infix operator |> { associativity left precedence 91 }
+
+public func |> <In, Out>(source: In, @noescape transform: In -> Out) -> Out {
+    return transform(source)
+}
+```
+
+or
+
+```
+infix operator ~> { associativity left precedence 91 }
+
+public func ~> <In, Out>(source: In, @noescape transform: In -> Out) -> Out {
+    return transform(source)
+}
+```
+
+and you can use them instead of `>-` operator.
+
+```swift
+let a /*: Observable<Int>*/ = Variable(1)
+let b /*: Observable<Int>*/ = Variable(2)
+
+combineLatest(a, b) { $0 + $1 }
+    |> filter { $0 >= 0 }
+    |> map { "\($0) is positive" }
+    |> subscribeNext { println($0) }
+```
+
+```swift
+let a /*: Observable<Int>*/ = Variable(1)
+let b /*: Observable<Int>*/ = Variable(2)
+
+combineLatest(a, b) { $0 + $1 }
+    ~> filter { $0 >= 0 }
+    ~> map { "\($0) is positive" }
+    ~> subscribeNext { println($0) }
+```
+
+So why was `>-` chosen in the end? Well, it was a difficult decision.
+
+Why wasn't standard function application operator used?
+
+I've first tried to find a similar operator in swift core libraries, but couldn't find it. That meant that I'll need to define something myself or find some third party library that contains reference function application operator definition and use it.
+Otherwise all of the example code would be unreadable.
+
+Why wasn't some standard library used for that operator?
+
+Well, I'm not sure there is a clear consensus in the community about funtion application operators or libraries that define them.
+
+Why wasn't function application operator defined only for `Observables` and `Disposables`?
```

Comment:
These questions should really be headings, so:

`#### Why wasn't function application operator defined only for `Observables` and `Disposables`?`